### PR TITLE
Delete nested tag inside codeFence

### PR DESCRIPTION
**Pull request by akamefi202**
Added a new rule to `htmlToMarkdownRules` to remove unnecessary nested tags inside `<pre>` tag.

### Fixed Issues
https://github.com/Expensify/App/issues/23659

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
In `ExpensiMark-Markdown-test.js` file, test named `Test codeFence copy from selection does not add extra new line`.
2. What tests did you perform that validates your changed worked?
Checked if `htmlToMarkdown` removes unnecessary tags correctly so extra newline character isn't added.

# QA
1. What does QA need to do to validate your changes?
Run the new test cases.
2. What areas to they need to test for regressions?
N/A

### CODE REVIEW

**Comment by akamefi202** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jasperhuangg** [comment]
@puneetlath assigning you since this is related to an issue you're managing

**Comment by mollfpr** [comment]
Sorry for the delay guys. I got cold yesterday 🙏

@akamefi202 You need to sign the commit https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#begin-coding-your-solution-in-a-pull-request

For the CLA, I guess you can comment `recheck` for the action to rerun.

**Comment by akamefi202** [comment]
recheck

File: package-lock.json
[the remark this answers is not in the document]
Comment (akamefi202) [reply package-lock.json:1]: Yes, I did.

File: __tests__/ExpensiMark-Markdown-test.js [hunk 1]
```diff
@@ -679,3 +679,14 @@ test('Test blockquote linebreak handling with text, block and inline elements',
...
 });
+
+test('Test codeFence copy from selection does not add extra new line', () => {
+    let testString = '<div><div><comment><pre><div><span>code</span><br></div></pre>text</comment></div></div>';
+    expect(parser.htmlToMarkdown(testString)).toBe('```\ncode\n```\ntext');
+
+    testString = '<pre>code<br></pre>text';
+    expect(parser.htmlToMarkdown(testString)).toBe('```\ncode\n```\ntext');
+
+    testString = '<h3>test heading</h3><div><pre class=\"notranslate\"><code class=\"notranslate\">Code snippet\n</code></pre></div><blockquote><p><a href=\"https://www.example.com\">link</a></p></blockquote>';
```
Comment (puneetlath) [inline __tests__/ExpensiMark-Markdown-test.js:690]: Out of curiosity, why are we using an `h3` in this test instead of an `h1`?

Comment (akamefi202) [reply __tests__/ExpensiMark-Markdown-test.js:690]: If we copy & paste selection from Github, heading is shown as h3 tag.

Comment (puneetlath) [reply __tests__/ExpensiMark-Markdown-test.js:690]: Ah cool, I see.

**Review by puneetlath — APPROVED** [review]

**Comment by puneetlath** [comment]
Looks good to me! @akamefi202 you'll next need to open a PR in the App repo to point to the updated expensify-common commit hash.

**Comment by akamefi202** [comment]
@puneetlath @mollfpr Thanks. I will create a PR in the App repo too!

**Comment by akamefi202** [comment]
@puneetlath @mollfpr I created the PR in App repo.
https://github.com/Expensify/App/pull/24549

**Pull request merged**